sdk/include/ddk/ntstrsafe.h

```diff
@@ -2534,33 +2540,48 @@ NTSTRSAFEAPI RtlStringVPrintfWorkerA(STRSAFE_LPSTR pszDest,size_t cchDest,STRSAF
     return Status;
 }
 
-NTSTRSAFEAPI RtlStringVPrintfWorkerW(STRSAFE_LPWSTR pszDest,size_t cchDest,STRSAFE_LPCWSTR pszFormat,va_list argList)
+NTSTRSAFEAPI RtlpStringVPrintfWorkerW(STRSAFE_LPWSTR pszDest, size_t cchDest, STRSAFE_LPCWSTR pszFormat, size_t* pcchDestNewLen, va_list argList)
 {
     NTSTATUS Status = STATUS_SUCCESS;
-    if (cchDest==0)
-        Status = STATUS_INVALID_PARAMETER;
-    else
+    int iRet;
+    size_t cchMax;
+    cchMax = cchDest - 1;
+    iRet = _vsnwprintf(pszDest, cchMax, pszFormat, argList);
+    if ((iRet < 0) || (((size_t) iRet) > cchMax))
     {
-        int iRet;
-        size_t cchMax;
-        cchMax = cchDest - 1;
-        iRet = _vsnwprintf(pszDest,cchMax,pszFormat,argList);
-        if ((iRet < 0) || (((size_t)iRet) > cchMax))
+        pszDest += cchMax;
+        *pszDest = L'\0';
+        Status = STATUS_BUFFER_OVERFLOW;
+    }
+    else
+        if (((size_t) iRet) == cchMax)
         {
             pszDest += cchMax;
             *pszDest = L'\0';
```

Comment:
UNICODE_NULL

Reply:
Same : This is part of original code. My intention was not to modify existing functions.  Shall I touch this?

Reply:
Ditto.

Reply:
Using `ANSI_NULL` or `UNICODE_NULL` requires including an NT header that defines them, which is something that neither we nor the original MS ntstrsafe.h do right now.
So let's not fix what isn't broken.